EapWebInterface/src/main/java/de/hpi/unicorn/application/pages/input/generator/GeneratePanel.java

```diff
@@ -146,34 +147,51 @@ protected Object load() {
+                item.add(new Label("attributeType", attributeInput.getAttributeType().getName()));
+                final Label attributeInputDescriptionLabel = new Label("attributeInputDescription", getAttributeInputDescription(attributeInput));
+                attributeInputDescriptionLabel.setOutputMarkupId(true);
+                item.add(attributeInputDescriptionLabel);
+                attributeInputs.add(attributeInput);
                 IModel<String> attributeInputModel = new Model<String>() {
                     @Override
                     public String getObject() {
-                        return attributeInput.get(attribute);
+                        int indexOfAttributeInput = attributeInputs.indexOf(attributeInput);
+                        return attributeInputs.get(indexOfAttributeInput).getInput();
                     }
                     @Override
                     public void setObject(String inputValue) {
-                        attributeInput.put(attribute, inputValue);
+                        int indexOfAttributeInput = attributeInputs.indexOf(attributeInput);
+                        attributeInputs.get(indexOfAttributeInput).setInput(inputValue);
                     }
                 };
 
-                TextField<String> inputField = new TextField<>("attributeInput", attributeInputModel);
-                inputField.add(AttributeValidator.getValidatorForAttribute(attribute));
+                final TextField<String> inputField = new TextField<>("attributeInput", attributeInputModel);
+                inputField.add(attributeInput.getAttributeInputValidator());
                 inputField.setLabel(new Model<String>(attribute.getName()));
+                inputField.setOutputMarkupId(true);
                 item.add(inputField);
-                item.add(new Label("attributeInputWarning", "")
-                        .setVisible(attributeDependencyManager.isDependentAttributeInAnyDependency(attribute)));
+                Boolean isDependentAttribute = attributeDependencyManager.isDependentAttributeInAnyDependency(attribute);
+                item.add(new Label("attributeInputWarning", "").setVisible(isDependentAttribute));
+
+                DropDownChoice<AttributeInput.ProbabilityDistributionEnum> methodDropDown = new DropDownChoice<>(
+                        "attributeInputMethodSelection",
+                        new PropertyModel<AttributeInput.ProbabilityDistributionEnum>(attributeInput, "selectedMethod"),
+                        attributeInput.getAvailableMethods());
+                methodDropDown.add(new AjaxFormComponentUpdatingBehavior("onChange") {
+                    @Override
+                    protected void onUpdate(AjaxRequestTarget target) {
+                        attributeInputDescriptionLabel.detachModels();
+                        attributeInputDescriptionLabel.setDefaultModel(getAttributeInputDescription(attributeInput));
```

Comment:
What would be the correct method here? Couldn't find any other method doing this.
The initial model is set in the constructor, which resolves to several "super" calls, ending in a method for setting the model that isn't accessible for us.